sdk/core/core-rest-pipeline/src/interfaces.ts

```diff
@@ -56,12 +56,43 @@ export type RequestBodyType =
   | string
   | null;
 
+/**
+ * An interface compatible with NodeJS's `http.AgentOptions`
+ *
+ */
+export type AgentOptions = SecureContextOptions;
+
+/**
+ * An interface compatible with NodeJS's `http.SecureContextOptions`
+ *
+ */
+export interface SecureContextOptions {
+  /**
+   * Optionally override the trusted CA certificates. Necessary only if the server uses a self-signed certificate
+   */
+  ca?: string | Buffer | Array<string | Buffer>;
+  /**
+   * Cert chains in PEM format. One cert chain should be provided per private key.
+   * Only necessary when using client certificate authentication
+   */
+  cert?: string | Buffer | Array<string | Buffer>;
+  /**
+   * Private keys in PEM format
+   * Only necessary when using client certificate authentication
+   */
+  key?: string | Buffer | Array<Buffer | Object>;
+}
+
 /**
  * An interface compatible with NodeJS's `http.Agent`.
  * We want to avoid publicly re-exporting the actual interface,
  * since it might vary across runtime versions.
  */
 export interface Agent {
+  /**
+   * Agent Options
+   */
+  options?: AgentOptions;
```

Comment:
Why do we need this? They can configure a `https.Agent` that has all the options they want without us needing to know/see it, right?

Reply:
Yup, removed these changes